### ISSUE TITLE
Android: Checked/Unchecked State does not announce

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30843]
## Description
Does not announce "checked"/"not checked" on focus of:

- [ ] Button #30959
- [ ] TextInput #30960
- [ ] Slider #30961
- [ ] Picker #30962
- [ ] ProgressBarAndroid #30963

Note: This is likely directly related to #30843, where the "selected" accessibilityState is equally non-functional on this same set of components.

## React Native version:
0.63

## Expected Behavior
Upon focus of an element with the "checked" accessibilityState, the screen reader should announce either "checked" or "not checked", or in the case of elements with the "switch" role, it should announce "on" or "off". Upon activation, if the state is changed, the new state should be announced.

## Snack
https://snack.expo.io/ncQPLFuOz

## Android Details
The "checked" accessibilityState should correspond to the "checked" property of the `AccessibilityNodeInfo`.  For it to function properly, the "checkable" property of the `AccessibilityNodeInfo` must also be set to true. Since this is working for most components, I believe this is correctly implemented, and the issue lies on the Javascript side for these individual components.

**Comment by blavalla** [comment]
This parent issue is no longer useful for tracking, see sub-issues for details.
